spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/AddFilesProcedure.java

```diff
@@ -196,11 +193,36 @@ private void importFileTable(
       String format,
       Map<String, String> partitionFilter,
       boolean checkDuplicateFiles,
-      PartitionSpec spec,
       int parallelism) {
+
+    org.apache.spark.sql.execution.datasources.PartitionSpec inferredSpec =
+        getInferredSpec(spark(), tableLocation);
+
+    List<String> sparkPartNames =
```

Comment:
I would use the SparkTableUtil method we just merged in the other pr. Feel free to modify it as needed

Reply:
Thanks, I refactored the part that's commonly needed to a public method. PTAL